specs/Checkbox.md

```diff
@@ -0,0 +1,388 @@
+| checkmarkIconProps   | IIconProps                                                  |                                                                                |
+| className            | string                                                      |                                                                                |
+| componentRef         | IRefObject<ICheckbox>                                       |                                                                                |
+| defaultChecked       | boolean                                                     |                                                                                |
+| defaultIndeterminate | boolean                                                     |                                                                                |
+| disabled             | boolean                                                     |                                                                                |
+| indeterminate        | boolean                                                     |                                                                                |
+| inputProps           | React.ButtonHTMLAttributes<HTMLElement or HTMLButtonElement>|                                                                                |
+| keytipProps          | IKpeytipProps                                               |                                                                                |  
+| label                | string                                                      |                                                                                |
+| onChange             | (ev, checked) => void                                       |                                                                                |
+| onRenderLabel        | IRenderFunction<ICheckboxProps>                             |                                                                                |
+| styles               | IStyleFunctionOrObject<ICheckboxStyleProps, ICheckboxStyles>|                                                                                |
+| theme                | ITheme                                                      |                                                                                |
+
+### Stardust Checkbox props
+
+
+| Name                 | Type                                                        | Notes                                                                          |
+| -------------------- | --------------------------------------------------------    | -------------------------------------------------------------------------------|
+| animation            | AnimationProp                                               |                                                                                |
```

Comment:
GROUP REVIEW:

This is solved by agreeing on component best practices.  This is a global shorthand prop on all components.  It allows `<Animation />` to be easily added to any component.  The same practice is followed for other components like `<Tooltip />` and `<Popup />`.  The idea is, make it easier to use components, doing things like animating should be easy.